trtools/dumpSTR/dumpSTR.py

```diff
@@ -514,102 +524,167 @@ def GetAllCallFilters(call_filters):
-                    sampdat.append("NOCALL")
-                else: sampdat.append(sample[key])
-            call = vcf.model._Call(record, sample.sample, samp_fmt(*sampdat))
-            new_samples.append(call)
+        sample_info['totaldp'][:] = np.nan
+
+    filtered_samples = np.logical_and(
+            all_filter_text != 'PASS', all_filter_text != 'NOCALL'
+    )
+    if not np.any(filtered_samples):
+        return record #nothing else to do
+    
+    # mask the filtered genotypes
+    ploidy = record.GetMaxPloidy()
+    for idx in filtered_samples.nonzero()[0]:
+        record.vcfrecord.genotypes[idx] = [-1]*ploidy + [False]
+    # This line isn't actually a no-op, see docs here:
+    # https://github.com/brentp/cyvcf2/blob/master/docs/source/writing.rst
+    record.vcfrecord.genotypes = record.vcfrecord.genotypes
+
+    # mask all other format fields
+    for field in record.format:
+        if field == 'GT' or field == 'FILTER':
             continue
-        filter_reasons = FilterCall(sample, call_filters)
-        if len(filter_reasons) > 0:
-            for r in filter_reasons:
-                sample_info[sample.sample][r] += 1
-            for i in range(len(samp_fmt._fields)):
-                key = samp_fmt._fields[i]
-                if key == "GT":
-                    sampdat.append("./.")
-                else:
-                    if key == "FILTER": sampdat.append(",".join(filter_reasons))
-                    else: sampdat.append(None)
+        vals = record.format[field]
+        # null the filtered values
+        # different null value for different array types
+        if vals.dtype.kind == 'U':
+            vals[filtered_samples] = '.'
+            vals = np.char.encode(vals)
+        elif vals.dtype.kind == 'f':
+            vals[filtered_samples] = np.nan
+        elif vals.dtype.kind == 'i':
+            vals[filtered_samples] = _NOCALL_INT_FORMAT_VAL
         else:
-            sample_info[sample.sample]["numcalls"] += 1
-            try:
-                sample_info[sample.sample]["totaldp"] += sample["DP"]
-            except AttributeError:
-                sample_info[sample.sample]["totaldp"] += sample["LC"] # EH has LC field
-            except AttributeError: pass # If no DP, or LC, no coverage data available
-            for i in range(len(samp_fmt._fields)):
-                key = samp_fmt._fields[i]
-                if key == "FILTER": sampdat.append("PASS")
-                else: sampdat.append(sample[key])
-        call = vcf.model._Call(record, sample.sample, samp_fmt(*sampdat))
-        new_samples.append(call)
-    record.samples = new_samples
-    return record
+            raise ValueError("Found an unexpected format dtype for"
+                             " format field " + field)
+        record.vcfrecord.set_format(field, vals)
+
+    # rebuild the TRRecord with the newly modified cyvcf2 vcfrecord
+    # TODO test that this works under all permuations of inputs
```

Comment:
Good catch.

I just ran code coverage, these branches are all covered (presumably by the test_output tests) except for the record.HasFabricatedRefAllele() section. That's not tested because I don't have an ExpansionHunter test because the EH filters still don't work. When someone fixes those and adds a test for it that'll get covered. As of now the rest of the lines there are tested and working!